src/mito2/src/wal.rs

```diff
@@ -60,13 +62,19 @@ impl<S: LogStore> Wal<S> {
             store: self.store.clone(),
             entries: Vec::new(),
             entry_encode_buf: Vec::new(),
+            namespaces: HashMap::new(),
         }
     }
 
     /// Scan entries of specific region starting from `start_id` (inclusive).
-    pub fn scan(&self, region_id: RegionId, start_id: EntryId) -> Result<WalEntryStream> {
+    pub fn scan<'a>(
```

Comment:
Question: I am not sure if we can remove the `'a` here. Looks like it's not necessary.

Reply:
If without the lifetime, the compiler reports an error: 
```rust
error: lifetime may not live long enough
  --> src/mito2/src/wal.rs:96:9
   |
71 |         & self,
   |         - let's call the lifetime of this reference `'2`
...
74 |         wal_options: &WalOptions,
   |                      - let's call the lifetime of this reference `'1`
...
96 |         Ok(Box::pin(stream))
   |         ^^^^^^^^^^^^^^^^^^^^ method was supposed to return data with lifetime `'2` but it is returning data with lifetime `'1`
   |
help: consider introducing a named lifetime parameter and update trait if needed
   |
70 ~     pub fn scan<'a>(
71 ~         &'a  self,
72 |         region_id: RegionId,
73 |         start_id: EntryId,
74 ~         wal_options: &'a WalOptions,
   |
```